stdlib/hashtbl.ml

```diff
@@ -352,23 +409,38 @@ module type S =
   sig
     type key
     type 'a t
-    val create: int -> 'a t
+    val create : int -> 'a t
     val clear : 'a t -> unit
     val reset : 'a t -> unit
-    val copy: 'a t -> 'a t
-    val add: 'a t -> key -> 'a -> unit
-    val remove: 'a t -> key -> unit
-    val find: 'a t -> key -> 'a
-    val find_all: 'a t -> key -> 'a list
+    val copy : 'a t -> 'a t
+    val add : 'a t -> key -> 'a -> unit
+    val remove : 'a t -> key -> unit
+    val find : 'a t -> key -> 'a
+    val find_all : 'a t -> key -> 'a list
     val replace : 'a t -> key -> 'a -> unit
     val mem : 'a t -> key -> bool
-    val iter: (key -> 'a -> unit) -> 'a t -> unit
+    val iter : (key -> 'a -> unit) -> 'a t -> unit
     val filter_map_inplace: (key -> 'a -> 'a option) -> 'a t -> unit
-    val fold: (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
-    val length: 'a t -> int
+    val fold : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
+    val length : 'a t -> int
     val stats: 'a t -> statistics
   end
 
+module type FULL =
```

Comment:
wouldn’t something like `SFull` be a more consistent name?

-

this sounds awkward and irregular. is it really necessary? i understand that might be needed not to break code claiming to have signature `S`. how common is that? can’t we ask implementors to adapt their code?

otherwise, we might find ourselves iterating this scheme after each addition to Hashtbl: `SFullFull`, `SFullFullFull` and so on.